app/services/service_catalog/service_plans.rb

```diff
@@ -0,0 +1,36 @@
+module ServiceCatalog
+  class ServicePlans
+    attr_reader :items
+    def initialize(portfolio_item_id)
+      @portfolio_item_id = portfolio_item_id
+    end
+
+    def process
+      result = api_instance.list_service_offering_service_plans(service_offering_ref)
+      filter_result(result)
+      self
+    rescue StandardError => e
+      Rails.logger.error("Service Plans #{e.message}")
+      raise
+    end
+
+    def service_offering_ref
+      PortfolioItem.find(@portfolio_item_id).service_offering_ref
+    end
+
+    def filter_result(result)
+      @items = result.each_with_object([]) do |obj, array|
```

Comment:
A method named filter_result should just filter and return the filtered list.  It should probably not have a side effect of setting @items.  I would expect the caller to do that.